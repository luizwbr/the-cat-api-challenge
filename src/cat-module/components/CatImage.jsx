import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const DEFAULT_WIDTH = 400;
const StyledDiv = styled.div`
    padding: 5px;
`;
const StyledImg = styled.img`
    width: ${DEFAULT_WIDTH}px;
    height: auto;
    background: #efefef;
`;

/**
 * CatImage Component
 *  - Renders an image of a cat
 *
 * @param {*} props
 * @param {string} props.url - Image url
 * @param {number} props.width - Image width

 * @returns string
 */
function CatImage(props) {
    const { id, url, width = DEFAULT_WIDTH } = props;

    return (
        <StyledDiv>
            <StyledImg id={`image-${id}`} src={url} width={width} loading="lazy" alt={`cat-image-${id}`} />
        </StyledDiv>
    );
}

CatImage.propTypes = {

    /**
     * id - unique id for the image
     */
    id: PropTypes.string.isRequired,

    /**
     * The url of the image
     */
    url: PropTypes.string.isRequired,

    /**
     * The width of the image
     */
    width: PropTypes.number,
};

export default CatImage;
