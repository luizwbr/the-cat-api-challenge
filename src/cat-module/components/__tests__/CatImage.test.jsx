import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import CatImage from '../CatImage.jsx';

const mockProps = {
    id: '6rb',
    url: 'https://cdn2.thecatapi.com/images/6rb.gif',
    width: 450,
};

test('render imagem correctly', async () => {    
    // eslint-disable-next-line max-len
    const { getByAltText } = await render(<CatImage id={mockProps.id} url={mockProps.url} width={mockProps.width} />);
    
    const image = getByAltText('cat-image-6rb');
    expect(image.src).toContain('https://cdn2.thecatapi.com/images/6rb.gif');
});

test('renders', () => {
    // eslint-disable-next-line max-len
    const { asFragment } = render(<CatImage id={mockProps.id} url={mockProps.url} width={mockProps.width} />);

    expect(asFragment()).toMatchSnapshot();
});

