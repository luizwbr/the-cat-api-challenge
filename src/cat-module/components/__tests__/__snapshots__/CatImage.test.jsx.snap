// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ iHeqKj"
  >
    <img
      alt="cat-image-6rb"
      class="sc-gsnTZi dUmWvC"
      id="image-6rb"
      loading="lazy"
      src="https://cdn2.thecatapi.com/images/6rb.gif"
      width="450"
    />
  </div>
</DocumentFragment>
`;
